### PR TITLE
Update "PyData Ecosystem" to "Scientific Python Ecosystem"

### LINKED ISSUE #3442: Use "PyData Ecosystem" or "Scientific Python Ecosystem"?
In the [project goals](https://github.com/GenericMappingTools/pygmt#project-goals), we mention that:

> Integration with the [PyData ecosystem](https://pydata.org/): numpy.ndarray or pandas.DataFrame for data tables, xarray.DataArray for grids, and geopandas.GeoDataFrame for geographical data.

After looking at the link https://pydata.org/, users (and even myself) may be still confused with the definition and scope of "PyData Ecosystem". The best explanation of "PyData Ecosystem" may be this post: https://stackoverflow.com/questions/18168400/the-pydata-ecosystem.

In addition to "PyData Ecosystem", a similar term is "Scientific Python Ecosystem". This term is used on the NumPy Nature paper (https://www.nature.com/articles/s41586-020-2649-2#Sec3 and the figure 2 https://www.nature.com/articles/s41586-020-2649-2/figures/2) and across the [scientific Python site](https://scientific-python.org/). The term is also well defined at https://tools.scientific-python.org/about/:

> The Scientific Python ecosystem is a loose federation of community developed and owned Python projects widely used in scientific research, technical computing, and data science.

I think we should use "Scientific Python Ecosystem" instead of "PyData Ecosystem".

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Using "Scientific Python ecosystem" instead of "PyData ecosystem" following the discussion in issue #3442.

Fixes #3442

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
Please also update this line:
https://github.com/GenericMappingTools/pygmt/blob/2454aa693993d31e0e9b86946827f07f53e99e18/examples/tutorials/advanced/date_time_charts.py#L194

File: README.md [hunk 1]
```diff
@@ -45,8 +45,8 @@ widely used across the Earth, Ocean, and Planetary sciences and beyond.
 - Build a Pythonic API for GMT.
 - Interface with the GMT C API directly using ctypes (no system calls).
 - Support for rich display in the Jupyter notebook.
-- Integration with the [PyData ecosystem](https://pydata.org/): `numpy.ndarray` or `pandas.DataFrame` for
-  data tables, `xarray.DataArray` for grids, and `geopandas.GeoDataFrame` for geographical data.
+- Integration with the [Scientific Python ecosystem](https://scientific-python.org/): `numpy.ndarray` or
```
Comment (seisman) [inline README.md:48]: I feel it should be "scientific Python ecosystem" following https://scientific-python.org/about/ and https://www.nature.com/articles/s41586-020-2649-2.

Comment (yvonnefroehlich) [reply README.md]: Yes, I was also wondering which version we want to use, as I found all three "Scientific Python Ecosystem", "Scientific Python ecosystem", and "scientific Python ecosystem". But going with the one used on the official website makes sense.

File: examples/tutorials/advanced/date_time_charts.py [hunk 2]
```diff
@@ -191,10 +191,10 @@
 #
 # In this example, instead of using a list of :class:`pandas.DatetimeIndex` objects,
 # ``x`` is initialized as an :class:`xarray.DataArray` object. This object provides a
-# wrapper around regular PyData formats. It also allows the data to have labeled
-# dimensions while supporting operations that use various pieces of metadata. The
-# following code uses :func:`pandas.date_range` to fill the DataArray with data, but
-# this is not essential for the creation of a valid DataArray.
+# wrapper around regular scientific Python formats. It also allows the data to have
```
Comment (seisman) [inline examples/tutorials/advanced/date_time_charts.py:194]: > This object provides a wrapper around regular scientific Python formats.

I'm unsure what "scientific Python formats" means. The official documentation (https://docs.xarray.dev/en/stable/generated/xarray.DataArray.html) says:

> DataArray provides a wrapper around numpy ndarrays that uses labeled dimensions and coordinates to support metadata aware operations.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/date_time_charts.py]: Good point!. Actually I was already a bit confused by "regular PyData formats". So, I just replaced the term. Maybe we should rewrite this sentence to something similar as in the official xarray docs, please see commit https://github.com/GenericMappingTools/pygmt/pull/3447/commits/27735247fb2a09dc3b58a4c0ef55ff2e7e4461fa.

**Review by seisman — APPROVED** [review]
Looks.good to me. I have updatedthe PR title to reflect the main changes.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**